lib/rules/prefer-const.js

```diff
@@ -96,6 +97,50 @@ function getDestructuringHost(reference) {
     return node;
 }
 
+/**
+ * Determines if a destructuring assignment node contains
+ * any MemberExpression nodes. This is used to determine if a
+ * variable that is only written once using destructuring can be
+ * safely converted into a const declaration.
+ * @param {ASTNode} node The ObjectPattern or ArrayPattern node to check.
+ * @returns {boolean} True if the destructuring pattern contains
+ *      a MemberExpression, false if not.
+ */
+function hasMemberExpressionAssignment(node) {
+    switch (node.type) {
+        case "ObjectPattern":
+            return node.properties.some(prop => {
+                if (prop) {
+                    const nodeToCheck = (prop.type === "RestElement")
```

Comment:
Probably `babel-eslint` makes the type `ExperimentalRestProeprty` even now. Null-check may be safer than...

Reply:
Good point, I'll update that.